python/pyiceberg/io/pyarrow.py

```diff
@@ -164,7 +163,27 @@ def to_input_file(self) -> "PyArrowFile":
         return self
 
 
+
 class PyArrowFileIO(FileIO):
+
+    def __init__(self, properties: Properties = {}):
+        self.get_fs_and_path: Callable = lru_cache(self._get_fs_and_path)
+        super().__init__(properties=properties)
+
+    def _get_fs_and_path(self, location: str) -> (FileSystem, str):
+        uri = urlparse(location)  # Create a ParseResult from the URI
+        if not uri.scheme:  # If no scheme, assume the path is to a local file
+            return FileSystem.from_uri(os.path.abspath(location))
+        elif uri.scheme in ['s3', 's3a', 's3n']:
```

Comment:
Nit:
```suggestion
        elif uri.scheme in {'s3', 's3a', 's3n'}:
```

Reply:
Does this change the "contains" to check in a set instead of list?  naively, I'd assume the list is cheaper to construct and search for small numbers of elements, but I'd be curious to better understand why this change?

Reply:
You're right on the performance:
```python
➜  iceberg git:(fd-add-docs) ✗ python3
Python 3.10.6 (main, Aug 30 2022, 04:58:14) [Clang 13.1.6 (clang-1316.0.21.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> f_set = f"""
... 's3a' in {'s3', 's3a', 's3n'}
... """
>>> 
>>> f_list = f"""
... 's3a' in ['s3', 's3a', 's3n']
... """
>>> 
>>> iters = 100000000
>>> 
>>> import timeit
>>> 
>>> print(f"Set: {timeit.timeit(stmt=f_set, number=iters)}")
print(f"List: {timeit.timeit(stmt=f_list, number=iters)}")

Set: 1.4491868750010326
>>> print(f"List: {timeit.timeit(stmt=f_list, number=iters)}")
List: 1.4320723749988247
```

For me it makes sense to convert it into a set because they elements are unique and there is no explicit sorting and therefore it makes more sense (to me) to convert it into a set.

Reply:
Interesting, the perf difference is really almost negligible (I had expected bigger).

either way, this isn't hot path and what's more important is stay consistent across codebase, so changing to set :)